src/crop.sol

```diff
@@ -137,17 +151,19 @@ contract CropJoin {
             require(int256(wad) > 0);
 
             require(gem.transferFrom(msg.sender, address(this), val));
-            vat.slip(ilk, urn, int256(wad));
+            vat.slip(ilk, urp, int256(wad));
 
             total = add(total, wad);
-            stake[urn] = add(stake[urn], wad);
+            stake[urp] = add(stake[urp], wad);
         }
-        crops[urn] = rmulup(stake[urn], share);
+        crops[urp] = rmulup(stake[urp], share);
         emit Join(val);
     }
 
     function exit(address guy, uint256 val) public virtual {
-        harvest(msg.sender, guy);
+        address urp = proxy[msg.sender];
+        require(urp != address(0), "CropJoin/no-urn-proxy");
```

Comment:
I think we should create the proxy in `exit` as well (and `flee`). If Emergency shutdown happens, most of users will have to first do `join` with 0 value to get it.

Reply:
They will have to `flux` the collateral to an UrnProxy they control and `tack` `stake` to it as well before they can `exit` or `flee` anyway--so I don't think it makes the experience any smoother, unless we also add some logic to `exit` so that it automatically gulps any `gem`+`stake` they have. I wouldn't be opposed to adding some convenience method like that separately for ES if you think it's important, but I don't want to put it all into `exit`/`flee`.